Simplenote/Classes/SPTagsListViewController.m

```diff
@@ -319,125 +262,231 @@ - (void)tableView:(UITableView *)tableView performAction:(SEL)action forRowAtInd
     }
 }
 
 - (NSIndexPath *)tableView:(UITableView *)tableView targetIndexPathForMoveFromRowAtIndexPath:(NSIndexPath *)sourceIndexPath toProposedIndexPath:(NSIndexPath *)proposedDestinationIndexPath {
-    if (sourceIndexPath.section != kSectionTags || proposedDestinationIndexPath.section != kSectionTags) {
-
+    if (sourceIndexPath.section != SPTagsListSectionTags || proposedDestinationIndexPath.section != SPTagsListSectionTags) {
         return sourceIndexPath;
     }
     
-    return proposedDestinationIndexPath ? proposedDestinationIndexPath : sourceIndexPath;
+    return proposedDestinationIndexPath ?: sourceIndexPath;
 }
 
-- (UITableViewCellEditingStyle)tableView:(UITableView *)tableView editingStyleForRowAtIndexPath:(NSIndexPath *)indexPath {
+- (BOOL)tableView:(UITableView *)tableView shouldIndentWhileEditingRowAtIndexPath:(NSIndexPath *)indexPath {
+    return indexPath.section == SPTagsListSectionTags;
+}
 
-    return self.bEditing ? UITableViewCellEditingStyleDelete : UITableViewCellEditingStyleNone;
+- (UITableViewCellEditingStyle)tableView:(UITableView *)tableView editingStyleForRowAtIndexPath:(NSIndexPath *)indexPath {
+    BOOL isTagRow = indexPath.section == SPTagsListSectionTags;
+    return self.bEditing && isTagRow ? UITableViewCellEditingStyleDelete : UITableViewCellEditingStyleNone;
 }
 
 - (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
-    
     if (editingStyle == UITableViewCellEditingStyleDelete) {
         [SPTracker trackTagRowDeleted];
 		[self removeTagAtIndexPath:indexPath];
     }
 }
 
 
+#pragma mark - Cell Setup
+
+- (void)configureCell:(SPTagListViewCell *)cell atIndexPath:(NSIndexPath *)indexPath {
+    switch (indexPath.section) {
+        case SPTagsListSectionSystem: {
+            [self configureSystemCell:cell atIndexPath:indexPath];
+            break;
+        }
+        case SPTagsListSectionTags: {
+            [self configureTagCell:cell atIndexPath:indexPath];
+            break;
+        }
+    }
+}
+
+- (void)configureSystemCell:(SPTagListViewCell *)cell atIndexPath:(NSIndexPath *)indexPath {
+    switch (indexPath.row) {
+        case SPTagsListSystemRowAllNotes: {
+            cell.tagNameTextField.text = NSLocalizedString(@"All Notes", nil);
+            cell.leftImage = [UIImage imageWithName:UIImageNameAllNotes];
+            break;
+        }
+        case SPTagsListSystemRowTrash: {
+            cell.tagNameTextField.text = NSLocalizedString(@"Trash-noun", nil);
+            cell.leftImage = [UIImage imageWithName:UIImageNameTrash];
+            break;
+        }
+        case SPTagsListSystemRowSettings: {
+            cell.tagNameTextField.text = NSLocalizedString(@"Settings", nil);
+            cell.leftImage = [UIImage imageWithName:UIImageNameSettings];
+            break;
+        }
+    }
+}
```

Comment:
¿Por qué todas las {} alrededor de las declaraciones de casos? :)  ¿Son necesarios?

Reply:
I REALLY should go thru our ObjC coding standard. It's been SO LONG!! Removing, thank you!